### ISSUE TITLE
pandas 0.23.0 causing error

### ISSUE THREAD

**Issue by rsrpsinr** [jupyter-incubator/sparkmagic#458]
Pandas 0.23.0 (released to pypi on May 16 2018) causing the following error:

```
The code failed because of a fatal error:
	Failed to register auto viz for notebook.
Exception details:
	"cannot import name 'DataError'".
```

Caused by scripts that contains the following import (changed in Pandas 0.23.0):
`from pandas.core.groupby import DataError`

https://github.com/jupyter-incubator/sparkmagic/search?utf8=%E2%9C%93&q=from+pandas.core.groupby+import+DataError&type=

**Comment by o-alex** [comment]
Is there any workarounds for this issue? I am also getting this error now.

**Comment by robzor92** [comment]
In pandas 0.23.0 the import path for DataError should be "**from pandas.core.base import DataError**"

**Comment by mauza** [comment]
I'm going to try and pull down the change from jimdowling and build from source. Is that the way to go if I want to get around this now and not wait for another release?

**Comment by mauza** [comment]
I guess I could just downgrade pandas.

**Comment by ivangvasquez** [comment]
This is indeed a brand new issue, as sparkmagic 0.12.5 working for us (anaconda 5.0.1 and 5.1.0). Attempted to downgrade pandas and causes all sorts of dependency problems. Can you suggest a temporary workaround? Install an earlier version of sparkmagic perhaps?

**Comment by ivangvasquez** [comment]
Easy enough:
pip install pandas==0.22.0

**Comment by jimdowling** [comment]
Another work around is to 
git clone https://github.com/logicalclocks/sparkmagic.git
cd sparkmagic
pip install --no-cache-dir ./hdijupyterutils 
pip install --no-cache-dir --upgrade ./autovizwidget
pip install --no-cache-dir ./sparkmagic

**Comment by kevcunnane** [comment]
@apetresc or @kkonrad I noticed you merged PRs recently. Can you please merge https://github.com/jupyter-incubator/sparkmagic/pull/459 from @jimdowling since it fixes this issue? It seems like pandas 0.22.0 doesn't work with Python 3.7, so we can't work at all with latest Python + SparkMagic.

Without this type of fix teams will have to fork / maintain private versions of sparkmagic to keep up to date on support, so it'd be great to see some of the community PRs accepted here. Thanks.

**Comment by kkonrad** [comment]
I'm sorry @kevcunnane but I don't have any way to do this. I only contributed this one PR and somehow it got in. To be honest I thought that project is no longer active judging by commit history and list of open PRs. But I don't know what is actual state.

**Comment by apetresc** [comment]
Hey guys - really sorry about the radio silence recently.

I'm running integration tests on these recent pandas and livy PRs as we speak.

**Comment by apetresc** [comment]
As an update, I've merged the PR with this fix, and a few others. I'm currently investigating if/how I can cut a release – I'm only recently added as a maintainer so I don't have all the credentials just yet. Hopefully I can upload the updated package to PyPI soon.

**Comment by boompig** [comment]
For me the workaround does not work, as it depends on a version of numpy that no longer compiles on my operating system...

**Comment by apetresc** [comment]
Which operating system is that, @boompig?

**Comment by boompig** [comment]
Mac OS X 10.13.5

**Comment by apetresc** [comment]
This should now be fixed in release 0.12.6!
